homeassistant/components/utility_meter/__init__.py

```diff
@@ -249,3 +256,22 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
     _LOGGER.info("Migration to version %s successful", config_entry.version)
 
     return True
+
+
+async def async_remove_device_old(hass: HomeAssistant, entry: ConfigEntry):
+    """Remove obsolete source entity devices."""
+    entity_registry = er.async_get(hass)
+    device_registry = dr.async_get(hass)
+
+    source_entity = entity_registry.async_get(entry.options[CONF_SOURCE_SENSOR])
+    for device_entry in dr.async_entries_for_config_entry(
```

Comment:
Why iterate over all device entries? You have `source_entity.device_id` and can just fetch that device directly?